pandas/core/indexes/multi.py

```diff
@@ -1391,6 +1391,9 @@ def copy(  # type: ignore[override]
 
     def __array__(self, dtype=None, copy=None) -> np.ndarray:
         """the array interface, return my values"""
+        if copy is True:
+            # Note: branch avoids `copy=None` for NumPy 1.x support
+            return np.array(self.values, dtype=dtype, copy=copy)
```

Comment:
```suggestion
```

This is unnecessary here, because `MultiIndex.values` is already a customly build array (see `def _values(self)` a bit higher up in this file)

Reply:
Ouch, right, in that case we need to raise for ``copy=False`!